tests/test_e2e_15_maintenance.py

```diff
@@ -120,3 +120,38 @@ def test_010_create_mw_on_switch_should_move_evc(self):
         h11.cmd('ip link del vlan100')
         h3.cmd('ip link del vlan100')
 
+    def test_011_delete_mw_on_switch(self):
+        # 0. Start maintenance window
+        start = datetime.now() + timedelta(seconds=60)
+        end = start + timedelta(seconds=60)
+        payload = {
+            "start": start.strftime(TIME_FMT),
+            "end": end.strftime(TIME_FMT),
+            "items": [
+                "00:00:00:00:00:00:02"
+            ]
+        }
+
+        api_url = KYTOS_API + '/maintenance/'
+        # 1 Send get request to API to get maintenance schema
+        response = requests.get(api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})["id"]
+        # 2. from the output of the GET request, extract the mw_id
+        json_data = response.json()
+        mw_id = json_data["id"]
+        # 3. Provide mw_id to API call to delete said mw_id
+        mw_api_url = KYTOS_API + '/maintenance/' + mw_id
+        delete_response = requests.delete(mw_api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})
+        # 4. Verify that code 204(success) is given back
+        assert delete_response.status_code == 204
+        # 5. Request list of maintenance windows again, and verify that the deleted mw_id is not on maintenance list
+        req = requests.get(api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})
+        json_data = req.json()
+        for attribute in json_data:
+            if attribute['id'] != mw_id:
+                print("Maintenance window deleted correctly")
```

Comment:
In a GET request you should not provide a payload.
You could also request for the specific maintenance window you are going to check, and see if the status is 404.
```suggestion
        req = requests.get(my_api_url)
        assert req.status_code == 404
```